test/inspect.test.js

```diff
@@ -82,314 +245,401 @@ test('inspect', function (t) {
+        'dummy local package skipped (editable)');
+
+      t.ok(pkg.dependencies['jinja2'] !== undefined, 'jinja2 found');
+      t.ok(pkg.dependencies['testtools'] !== undefined, 'testtools found');
+
+      t.end();
     });
+
+    t.end();
+  });
 });
 
-function pipInstall() {
-  return subProcess.execute('pip',
-    ['install', '-r', 'requirements.txt', '--disable-pip-version-check']);
-}
+test('inspect Pipfile with pinned versions', function (t) {
+  return Promise.resolve().then(function () {
+    chdirWorkspaces('pipfile-pipapp-pinned');
+    t.teardown(testUtils.activateVirtualenv('pip-app'));
+  })
+  .then(function () {
+    return plugin.inspect('.', 'Pipfile')
+  })
+  .then(function (result) {
+    var pkg = result.package;
+
+    t.test('package dependencies', function (t) {
+      Object.keys(pipAppExpectedDependencies).forEach(function (depName) {
+        t.match(
+          pkg.dependencies[depName],
+          pipAppExpectedDependencies[depName].data,
+          pipAppExpectedDependencies[depName].msg
+        );
+      });
+
+      t.end();
+    });
+
+    t.end();
+  });
+});
+
+var pipenvAppExpectedDependencies = {
+  'python-etcd': {
+    data: {
+      name: 'python-etcd',
+      version: /^0\.4/,
+    },
+    msg: 'python-etcd1 found with version >=0.4,<0.5',
+  },
+  jinja2: {
+    data: {
+      name: 'jinja2',
+      version: /^0|1|2\.[0-6]/,
+    },
+    msg: 'jinja2 found with version <2.7',
+  },
+  testtools: {
+    data: {
+      name: 'testtools',
+    },
+    msg: 'testtools found',
+  },
+};
+
+test('inspect pipenv app with user-created virtualenv', function (t) {
+  return Promise.resolve().then(function () {
+    chdirWorkspaces('pipenv-app');
+
+    var venvCreated = testUtils.ensureVirtualenv('pipenv-app');
+    t.teardown(testUtils.activateVirtualenv('pipenv-app'));
+    if (venvCreated) {
+      return testUtils.pipenvInstall();
+    }
+  })
+  .then(function () {
+    return plugin.inspect('.', 'Pipfile')
+  })
+  .then(function (result) {
+    var pkg = result.package;
+
+    t.test('package dependencies', function (t) {
+      Object.keys(pipenvAppExpectedDependencies).forEach(function (depName) {
+        t.match(
+          pkg.dependencies[depName],
+          pipenvAppExpectedDependencies[depName].data,
+          pipenvAppExpectedDependencies[depName].msg
+        );
+      });
+
+      t.end();
+    });
+
+    t.end();
+  });
+});
+
+test('inspect pipenv app with auto-created virtualenv', function (t) {
+  return Promise.resolve().then(function () {
+    chdirWorkspaces('pipenv-app');
+
+    // Use several teardown callbacks, called in reverse order.
+    var teardowns = [];
+    t.teardown(function () {
+      while (teardowns.length > 0) {
+        teardowns.pop()();
+      }
+    });
+
+    // Deactivate current virtualenv if active.
+    //
+    // Note: The pipenv execuable within the virtualenv will still be used to
+    // run pipenv despite the virtualenv being deactivated.
```

Comment:
This is for the case the user had `virtualenv` activated before running the tests? if so, I think it's better to fail the test in such a case instead of deactivating user's `virtualenv` ...

Reply:
If forgot to update this, it's no longer true...